### ISSUE TITLE
Error "is maps/stacktrace map type enabled in your kernel?" inside virtual machine

### ISSUE THREAD

**Issue by Sh4pe** [iovisor/bcc#1714]
I'm trying to get `bcc` working inside a virtual machine since my host is running macOS and there is no `bcc` on macOS.

So I'm trying to define and run a virtual machine using [Vagrant](https://www.vagrantup.com) which I can use to work with `bcc`. I'm defining the virtual machine with these settings:

* [Vagrantfile](https://gist.github.com/Sh4pe/3d761e0aa57f38ca45a766b8661bdc06#file-vagrantfile), which is using
* [setup-cargo.sh](https://gist.github.com/Sh4pe/3d761e0aa57f38ca45a766b8661bdc06#file-setup-cargo-sh) (containing the Ubuntu setup instructions for `bcc` which I've found in [INSTALL.md](https://github.com/iovisor/bcc/blob/master/INSTALL.md#ubuntu---binary) and
* [setup-kernel.sh](https://gist.github.com/Sh4pe/3d761e0aa57f38ca45a766b8661bdc06#file-setup-kernel-sh) (which I've found in `iovisor`'s [xdp-vagrant](https://github.com/iovisor/xdp-vagrant/blob/master/setup-kernel.sh)).

When I'm running `bcc`, I get these errors:

```
vagrant@ubuntu-xenial:~$ sudo /usr/share/bcc/tools/stackcount -p 5406 -U c:malloc
/virtual/main.c:13:1: error: could not open bpf map: Invalid argument
is maps/stacktrace map type enabled in your kernel?
BPF_STACK_TRACE(stack_traces, 1024);
^
/virtual/include/bcc/helpers.h:186:3: note: expanded from macro 'BPF_STACK_TRACE'
  BPF_TABLE("stacktrace", int, struct bpf_stacktrace, _name, _max_entries);
  ^
/virtual/include/bcc/helpers.h:59:76: note: expanded from macro 'BPF_TABLE'
#define BPF_TABLE(_table_type, _key_type, _leaf_type, _name, _max_entries) \
                                                                           ^
/virtual/include/bcc/helpers.h:55:4: note: expanded from macro '\
BPF_F_TABLE'
}; \
   ^
/virtual/main.c:21:41: error: bpf_table stack_traces failed to open
                    key.user_stack_id = stack_traces.get_stackid(
                                        ^
2 errors generated.
Failed to compile BPF module
```

I've checked that the kernel inside the VM has the right kernel configuration that is described [here](https://github.com/iovisor/bcc/blob/master/INSTALL.md#kernel-configuration) in `INSTALL.md`:

```
vagrant@ubuntu-xenial:~$ grep BPF /boot/config-4.4.0-121-generic
CONFIG_BPF=y
CONFIG_BPF_SYSCALL=y
CONFIG_BPF_JIT_ALWAYS_ON=y
CONFIG_NETFILTER_XT_MATCH_BPF=m
CONFIG_NET_CLS_BPF=m
CONFIG_NET_ACT_BPF=m
CONFIG_BPF_JIT=y
CONFIG_HAVE_BPF_JIT=y
CONFIG_HAVE_EBPF_JIT=y
CONFIG_BPF_EVENTS=y
CONFIG_TEST_BPF=m
```

I'm at a loss what I can do to get `bcc` running in the VM. Can someone please help me?

**Comment by goldshtn** [comment]
I think BPF stack trace support is only available since Linux 4.6, and you
seem to be running 4.4.

**Comment by Sh4pe** [comment]
Indeed, you are right. Using a kernel newer than 4.6 solves my problem.
